x-pack/plugins/lens/public/app_plugin/mounter.tsx

```diff
@@ -243,10 +243,12 @@ export async function mountApp(
       useEffect(() => {
         (async () => {
           const hasUserDataView = await data.dataViews.hasData.hasUserDataView().catch(() => false);
-          const hasEsData = await data.dataViews.hasData.hasESData().catch(() => true);
-          if (!hasUserDataView || !hasEsData) {
-            setEditorState('no_data');
-            return;
+          if (!hasUserDataView) {
+            const hasEsData = await data.dataViews.hasData.hasESData().catch(() => true);
+            if ((!hasUserDataView && hasEsData) || !hasEsData) {
```

Comment:
If this code is executed, `hasUserDataView` is always `false`, so the condition will always be truthy no matter what `hasEsData` will return - this means we can simply this logic and simply show the "no data" page if the user doesn't have a data view.